provider/datasource_sites.go

```diff
@@ -0,0 +1,213 @@
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's redactions",
+						},
+						"requests_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's requests",
+						},
+						"suspicious_ips_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's suspicious IPs",
+						},
+						"top_attacks_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's top attacks",
+						},
+						"whitelist_uri": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Reference to the site's whitelist",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func readSites(d *schema.ResourceData, m any) error {
+	pm := m.(providerMetadata)
+	sc := pm.Client
+	corp := pm.Corp
+
+	// API documentation:
+	// https://docs.fastly.com/signalsciences/api/#_corps__corpName__sites_get
+	sites, err := sc.ListSites(corp)
+	if err != nil {
+		return err
+	}
+
+	d.SetId("list_sites")
+
+	filter := d.Get("filter").(string)
+
+	return d.Set("sites", flattenSites(sites, filter))
+}
+
+// flattenSites models data into format suitable for saving to Terraform state.
+func flattenSites(data []sigsci.Site, filter string) []map[string]any {
+	results := []map[string]any{}
+	if len(data) == 0 {
+		return results
+	}
+
+	for _, site := range data {
+		data := map[string]any{
+			"agent_anon_mode":      site.AgentAnonMode,
+			"agent_level":          site.AgentLevel,
+			"agents_uri":           site.Agents["uri"],
+			"alerts_uri":           site.Alerts["uri"],
+			"analytics_events_uri": site.AnalyticsEvents["uri"],
+			"blacklist_uri":        site.Blacklist["uri"],
+			"block_duration_secs":  site.BlockDurationSeconds,
+			"block_http_code":      site.BlockHTTPCode,
+			"block_redirect_url":   site.BlockRedirectURL,
+			"created":              site.Created.String(),
+			"display_name":         site.DisplayName,
+			"events_uri":           site.Events["uri"],
+			"header_links_uri":     site.HeaderLinks["uri"],
+			"integrations_uri":     site.Integrations["uri"],
+			"members_uri":          site.Members["uri"],
+			"monitors_uri":         site.Monitors["uri"],
+			"name":                 site.Name,
+			"redactions_uri":       site.Redactions["uri"],
+			"requests_uri":         site.Requests["uri"],
+			"suspicious_ips_uri":   site.SuspiciousIPs["uri"],
+			"top_attacks_uri":      site.TopAttacks["uri"],
+			"whitelist_uri":        site.Whitelist["uri"],
+		}
+
+		// Prune any empty values that come from the default string value in structs.
+		for k, v := range data {
+			if v == "" {
+				delete(data, k)
+			}
+		}
+		results = append(results, data)
+	}
+
+	if filter != "" {
+		for idx, site := range results {
+			if site["name"] == filter || site["display_name"] == filter {
+				return results[idx : idx+1]
```

Comment:
The implementation for the `filter` attribute is 'match only one', e.g. the first site that matches is returned.

If we want the filter to return multiple potential sites, then the implementation should be updated.